source/server/configuration_impl.h

```diff
@@ -37,10 +40,42 @@ class NetworkFilterConfigFactory {
 public:
   virtual ~NetworkFilterConfigFactory() {}
 
-  virtual NetworkFilterFactoryCb tryCreateFilterFactory(NetworkFilterType type,
-                                                        const std::string& name,
-                                                        const Json::Object& config,
-                                                        Server::Instance& server) PURE;
+  /**
+  * Create a particular network filter factory implementation.  If the creation fails, an error will
+  * be thrown.  The returned callback should always be valid.
+  */
+  virtual NetworkFilterFactoryCb createFilterFactory(NetworkFilterType type,
```

Comment:
i think we are a bit more strict now around documenting @params etc, can you add those